hibernate-core/src/main/java/org/hibernate/internal/util/collections/ArrayHelper.java

```diff
@@ -428,13 +407,8 @@ public static int countNonNull(Serializable[] array) {
 	}
 
 	public static String[] reverse(String[] source) {
-		final int length = source.length;
-		final String[] destination = new String[length];
-		for ( int i = 0; i < length; i++ ) {
-			final int x = length - i - 1;
-			destination[x] = source[i];
-		}
-		return destination;
+		Collections.reverse( Arrays.asList( source ) );
```

Comment:
If we change this, there will be a new `List` object created just for reversing the array. The Java version also created more garbage internally. This change also adds a side-effect. The original method creates a new array, and doesn't modify the source. This one will modify the source array.